### ISSUE TITLE
Cannot open graphic device on MacOSX

### ISSUE THREAD

**Issue by thorstenwagner** [s-u/Rserve#84]
I've established a working RConnection through RServe on MacOSX.
```
c.eval("print(5)");
c.eval("png(filename=\"abx.png\")");
```
However, the first command runs nicely, the second fails with the error message 
"org.rosuda.REngine.Rserve.RserveException: eval failed" That's the only information about the problem that I have.

Both commands are working on my local R installation.

On windows it is working as well, so it seems to be MacOSX specific.

Any advice?

**Comment by hyounesy** [comment]
I also have similar issue with all bitmap devices (png, jpeg, bmp, tiff) on OSX. For now I am using svg and pdf devices, then when I need bitmap, I use a convertor to go from svg to png. But would be a great help if I could get it working out of the box in R.

**Comment by s-u** [comment]
I can't comment on the particular issue without more details, but we're using the `Cairo` device in `Rserve` applications and it has never caused any issues. The problem with the built-in R devices is that depending on the built they often rely on interactive sessions which is not available in Rserve as you're typically running a headless service and not an interactive session.

**Comment by hyounesy** [comment]
I personally am using Rserve in combination with java (REngine). I just found out that this particular issue only happens whenever I start Rserve from the java application using the `launchRserve()` in StartRserve.java. This is the error message I get as soon as I try to plot to a png device:
```
Rserve> The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Rserve> Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
```

However, if I start the Rserve manually either within a normal interactive R session, or directly from terminal using R CMD, and then connect with the java client there is no issue and can create a png plot.
```
> /Library/Frameworks/R.framework/Resources/bin/R CMD /Library/Frameworks/R.framework/Versions/3.4/Resources/library/Rserve/libs//Rserve --no-save --slave 
```
I also tested with debug = TRUE and FALSE. Neither works in the first case, but both work fine in the second case.
